packages/tools-iottwinmaker/src/lib/workspace.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  CreateWorkspaceCommandOutput,
+  GetWorkspaceCommandOutput,
+  ResourceNotFoundException,
+} from '@aws-sdk/client-iottwinmaker';
+
+import { createUniqueId, replaceTemplateVars, retryFunction } from './utils';
+import { getDefaultAwsClients as aws } from './aws-clients';
+
+import { WORKSPACE_ROLE_ASSUME_POLICY } from './policy/workspace-role-assume-policy';
+import { WORKSPACE_DASHBOARD_ROLE_ASSUME_POLICY } from './policy/workspace-dashboard-role-assume-policy';
+import { workspaceRolePolicyTemplate } from './policy/workspace-role-policy';
+import { workspaceDashboardRolePolicyTemplate } from './policy/workspace-dashboard-role-policy';
+import { CreatePolicyCommandOutput, CreateRoleCommandOutput } from '@aws-sdk/client-iam';
+import { CreateBucketCommandOutput } from '@aws-sdk/client-s3';
+
+/**
+ * Helper function during workspace creation to create and attach roles and policies
+ * @param uniqueId identifier for the role and policy used as suffix in name
+ * @param roleType Role or DashboardRole
+ * @param roleAssumePolicy stringified JSON of the policy
+ * @param rolePolicy stringified JSON of the role
+ * @returns promise of the roleArn
+ */
+async function createRoleAndPolicy(
+  uniqueId: string,
+  roleType: 'WorkspaceRole' | 'WorkspaceDashboardRole',
+  roleAssumePolicy: string,
+  rolePolicy: string
+) {
+  const roleName = `IoTTwinMaker${roleType}-${uniqueId}`;
+  console.log(`Creating Role ${roleName}...`);
+  const createRoleResponse: CreateRoleCommandOutput = await aws().iam.createRole({
+    RoleName: roleName,
+    Path: '/',
+    AssumeRolePolicyDocument: roleAssumePolicy,
+    Description: `${roleType} generated by AWS IoT TwinMaker Getting Started Script`,
+  });
+
+  if (!createRoleResponse.Role || !createRoleResponse.Role.Arn) {
+    throw new Error(`Failed to create role ${roleName}.`);
+  }
+
+  const roleArn: string = createRoleResponse.Role.Arn;
+  console.log(`Created Role: ${roleArn}.`);
+  const createPolicyResponse: CreatePolicyCommandOutput = await aws().iam.createPolicy({
+    PolicyName: `IoTTwinMaker${roleType}Policy-${uniqueId}`,
+    PolicyDocument: rolePolicy,
+    Description: `${roleType} Policy generated by AWS IoT TwinMaker Getting Started Script`,
+  });
+
+  const policyArn: string | undefined = createPolicyResponse.Policy?.Arn;
```

Comment:
Above you throw an error if the role is not created successfully, should follow same pattern here